visivo/models/base/eval_string.py

```diff
@@ -0,0 +1,131 @@
+import os
+from typing import Any, List
+import ast
+import operator as op
+import re
+from visivo.models.base.context_string import ContextString
+
+INLINE_CONTEXT_STRING_REGEX = r"\${\s*ref\([a-zA-Z0-9\s'\"\-_]+?\)\s*}"
+EVAL_STRING_REGEX = r"^>{(.*)}$"
+
+"""
+EvalString represents a string that contains expressions to be evaluated.
+
+This class is used for handling strings that may contain references or expressions
+enclosed in '>{' and '}' delimiters. It provides functionality to validate the
+string format and extract embedded ContextString instances.
+
+Available injected methods:
+    - any_test_failed() -> bool
+
+Available injected attributes:
+    - env -> Dict[str, Any]
+"""
+
+
+class EvalString:
+    def __init__(self, value: str):
+        self.value = value
+
+    def __str__(self):
+        return self.value
+
+    def get_context_strings(self) -> List[ContextString]:
+        return list(
+            map(
+                lambda m: ContextString(m),
+                re.findall(INLINE_CONTEXT_STRING_REGEX, self.value),
+            )
+        )
+
+    def evaluate(self) -> Any:
```

Comment:
Is the idea here to make it safer than just running `eval()` around the string?

Reply:
Yes, it can only execute permitted operations